### PR TITLE
feat: Add notifications for spending limit txs

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

- Shows notifications when executing a spending limit tx
- Removes `tx` from `TxEvent` types as it is currently not utilized
- Adds an optional `message` prop to `TxEvents`

### CODE REVIEW

**Comment by iamacook** [comment]
It might make sense to simply add a new `TxEvent` for spending limit executions. It then wouldn't be necessary to pass a `message` with the dispatched action. What do you think?

**Comment by katspaugh** [comment]
I agree with Aaron, it should be a new event.

**Comment by katspaugh** [comment]
Since you're at it, could you also replace `mining` in the notification messages with `being processed` (so that it works for PoS chains too).

**Comment by usame-algan** [comment]
> Since you're at it, could you also replace mining in the notification messages with being processed (so that it works for PoS chains too).

I renamed the Events to `PROCESSING`, `PROCESSING_MODULE` and `PROCESSED`

File: src/hooks/useTxNotifications.ts [hunk 1]
```diff
@@ -6,15 +6,17 @@ import { AppRoutes } from '@/config/routes'
...
   [TxEvent.PROPOSE_FAILED]: 'Failed proposing the transaction. Please try again.',
   [TxEvent.SIGNATURE_PROPOSED]: 'You successfully signed the transaction.',
   [TxEvent.SIGNATURE_PROPOSE_FAILED]: 'Failed to send the signature. Please try again.',
   [TxEvent.EXECUTING]: 'Please confirm the execution in your wallet.',
-  [TxEvent.MINING]: 'Your transaction is being mined.',
-  [TxEvent.MINED]: 'Your transaction was successfully mined and is now being indexed.',
+  [TxEvent.PROCESSING]: 'Your transaction is being mined.',
+  [TxEvent.PROCESSING_MODULE]:
+    'Your transaction has been submitted and will appear in the interface only after it has been successfully mined and indexed.',
+  [TxEvent.PROCESSED]: 'Your transaction was successfully mined and is now being indexed.',
```
Comment (iamacook) [inline src/hooks/useTxNotifications.ts:19]: Maybe we should adjust these to say "mined/validated"?

**Review by iamacook — APPROVED** [review]
Looks good. We might want to adjust the notification text slightly for those that are "processing/-ed" though.

File: src/hooks/useTxNotifications.ts (hunk 1 above)
Comment (katspaugh) [reply src/hooks/useTxNotifications.ts:19]: When I asked to change the wording, I meant the notification message itself, not the enum props.

Comment (usame-algan) [reply src/hooks/useTxNotifications.ts:19]: Lets leave the enum naming how it was and only update the messages.

Comment (usame-algan) [reply src/hooks/useTxNotifications.ts]: done

**Review by iamacook — APPROVED** [review]

File: src/services/tx/txEvents.ts [hunk 2]
```diff
@@ -12,26 +10,29 @@ export enum TxEvent {
...
-  [TxEvent.SUCCESS]: { txId: string; batchId?: string }
+  [TxEvent.SIGNED]: { txId?: string }
+  [TxEvent.SIGN_FAILED]: { txId?: string; error: Error }
+  [TxEvent.PROPOSE_FAILED]: { error: Error }
+  [TxEvent.PROPOSED]: { txId: string }
+  [TxEvent.SIGNATURE_PROPOSE_FAILED]: { txId: string; error: Error }
+  [TxEvent.SIGNATURE_PROPOSED]: { txId: string }
+  [TxEvent.EXECUTING]: Id
+  [TxEvent.MINING]: Id & { txHash: string }
+  [TxEvent.MINING_MODULE]: Id & { txHash: string; message: string }
```
Comment (katspaugh) [inline src/services/tx/txEvents.ts:31]: We agreed to remove the message right?

File: src/services/tx/txSender.ts [hunk 3]
```diff
@@ -230,84 +229,109 @@ export const dispatchBatchExecution = async (
...
+  } catch (error) {
+    txDispatch(TxEvent.FAILED, { groupKey: id, error: error as Error })
+    throw error
+  }
+
+  txDispatch(TxEvent.MINING_MODULE, {
+    groupKey: id,
+    txHash: result.hash,
+    message:
+      'Your transaction has been submitted and will appear in the interface only after it has been successfully mined and indexed.',
```
Comment (katspaugh) [inline src/services/tx/txSender.ts:320]: Rm the message please :)

**Review by katspaugh — CHANGES_REQUESTED** [review]

File: src/services/tx/txEvents.ts (hunk 2 above)
Comment (usame-algan) [reply src/services/tx/txEvents.ts:31]: Ah, good catch! Forgot to remove it after resetting the commit

**Review by katspaugh — APPROVED** [review]
🚀

**Pull request merged**